libs/object/tests/refCountedObjectTest.cpp

```diff
@@ -430,7 +430,8 @@ struct AsyncDeleter
 		try
 		{
 			// Ideally we want to show here how to use dispatch queues, but we cannot add DispatchQueue liblet dependency here.
-			std::async(std::launch::async, [obj]() noexcept { TObject::RefCountPolicy::template Delete(obj); });
+			auto x = std::async(std::launch::async, [obj]() noexcept { TObject::RefCountPolicy::template Delete(obj); });
+			x.wait();
```

Comment:
This causes the object to be destroyed synchronously that defeats the purpose of the unit test.
To avoid the compilation error, please add (void) in front of the std::async call instead:
(void)std::async(...)

The same is for the other change.

Reply:
I reverted to the original code, and instrumented the test to see what was actually happening. It wasn't working as you describe. std::async returns a std::future, which is immediately destroyed. The future dtor block until the async operation completes, as described [here](https://en.cppreference.com/w/cpp/thread/future/~future).

This is why I put in the wait call.

If you'd like I can change the test to use std::thread(functor).detach() which will let it run in the background and not block when the thread object destructs.

Reply:
Ouch... Yes, please do the change to use the thread detach then. The goal of the test was to have the asynchronous destruction.